### PR TITLE
[REFACTOR] Kibana and ECK will have their own renderer

### PR DESCRIPTION
**Pull request by asincu**
## Description

This PR proposes to split logstorage.go into three renderers:
-  A renderer for Kibana  (this will live in a new sub-package in logstorage package)
- A renderer for ECK operator (this will live in a new sub-package in logstorage package)
- A renderer for Elasticsearch and other other components (this will remain in logstorage.go)

This is the first step to introduce changes for Kibana per tenant.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/logstorage.go [hunk 1]
```diff
@@ -354,37 +264,13 @@ func (es *elasticsearchComponent) Objects() ([]client.Object, []client.Object) {
...
-		toCreate = append(toCreate, networkpolicy.AllowTigeraDefaultDeny(KibanaNamespace))
-		toCreate = append(toCreate, es.kibanaServiceAccount())
-
-		if len(es.cfg.PullSecrets) > 0 {
-			toCreate = append(toCreate, secret.ToRuntimeObjects(secret.CopyToNamespace(KibanaNamespace, es.cfg.PullSecrets...)...)...)
-		}
-
-		if len(es.kibanaSecrets) > 0 {
-			toCreate = append(toCreate, secret.ToRuntimeObjects(es.kibanaSecrets...)...)
+	if es.cfg.KeyStoreSecret != nil {
```
Comment (tmjd) [inline pkg/render/logstorage.go:267]: This use to only be done when KibanaEnabled was false. Did you figure out it is ok to do this when KibanaEnabled is true?

File: pkg/render/logstorage/kibana/kibana.go [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+}
+
+func (k *kibana) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (k *kibana) Objects() ([]client.Object, []client.Object) {
+	var toCreate, toDelete []client.Object
+
+	// Doesn't matter what the cluster type is, if LogStorage exists and the DeletionTimestamp is set finalized the
```
[the remark this answers is not in the document]
Comment (asincu) [reply pkg/render/logstorage/kibana/kibana.go:144]: Copy Paste. Will change it.

File: pkg/render/logstorage.go (hunk 1 above)
Comment (asincu) [reply pkg/render/logstorage.go:267]: I think this was only enabled when FIPS was enabled. And for FIPS we had Kibana disabled. 

See the code in controller. 
https://github.com/tigera/operator/blob/master/pkg/controller/logstorage/elastic/elastic_controller.go#L405-L423

**Review by tmjd — APPROVED** [review]
LGTM, please make sure you squash the commits before or as part of the merge.

**Pull request merged**